application/MultiMC.cpp

```diff
@@ -272,6 +276,91 @@ MultiMC::MultiMC(int &argc, char **argv) : QApplication(argc, argv)
+    QString originalData = fi.absolutePath();
+    // if the config file exists in Contents/MacOS, then user data is still there and needs to moved
+    if (QFileInfo::exists(FS::PathCombine(originalData, "multimc.cfg")))
+    {
+        qDebug() << "On macOS and found config file in old location, moving user data...";
+        QDir dir;
+        // if the warnings are hit then the original path likely doesn't exist
+        if (!dir.rename(FS::PathCombine(originalData, "instances"), FS::PathCombine(dataPath, "instances")))
+        {
+            qWarning() << "Failed to move instances folder";
+        }
+        if (!dir.rename(FS::PathCombine(originalData, "accounts"), FS::PathCombine(dataPath, "accounts")))
+        {
+            qWarning() << "Failed to move accounts folder";
+        }
+        if (!dir.rename(FS::PathCombine(originalData, "assets"), FS::PathCombine(dataPath, "assets")))
+        {
+            qWarning() << "Failed to move assets folder";
+        }
+        if (!dir.rename(FS::PathCombine(originalData, "icons"), FS::PathCombine(dataPath, "icons")))
+        {
+            qWarning() << "Failed to move icons folder";
+        }
+        if (!dir.rename(FS::PathCombine(originalData, "libraries"), FS::PathCombine(dataPath, "libraries")))
+        {
+            qWarning() << "Failed to move libraries folder";
+        }
+        if (!dir.rename(FS::PathCombine(originalData, "mods"), FS::PathCombine(dataPath, "mods")))
+        {
+            qWarning() << "Failed to move mods folder";
+        }
+        if (!dir.rename(FS::PathCombine(originalData, "meta"), FS::PathCombine(dataPath, "meta")))
+        {
+            qWarning() << "Failed to move meta folder";
+        }
+        if (!dir.rename(FS::PathCombine(originalData, "themes"), FS::PathCombine(dataPath, "themes")))
+        {
+            qWarning() << "Failed to move themes folder";
+        }
+        if (!dir.rename(FS::PathCombine(originalData, "translations"), FS::PathCombine(dataPath, "translations")))
+        {
+            qWarning() << "Failed to move translations folder";
+        }
+        if (!dir.rename(FS::PathCombine(originalData, "cache"), FS::PathCombine(dataPath, "cache")))
+        {
+            qWarning() << "Failed to move cache folder";
+        }
+        if (!dir.rename(FS::PathCombine(originalData, "metacache"), FS::PathCombine(dataPath, "metacache")))
+        {
+            qWarning() << "Failed to move metacache";
+        }
+        if (!dir.rename(FS::PathCombine(originalData, "multimc.cfg"), FS::PathCombine(dataPath, "multimc.cfg")))
+        {
+            qWarning() << "Failed to move multimc.cfg";
+        }
+        if (!dir.rename(FS::PathCombine(originalData, "accounts.json"), FS::PathCombine(dataPath, "accounts.json")))
+        {
+            qWarning() << "Failed to move accounts.json";
+        }
+        if (!dir.rename(FS::PathCombine(originalData, "MultiMC-0.log"), FS::PathCombine(dataPath, "MultiMC-0.log")))
```

Comment:
Not sure if you really need to move the logs, but if you do, what about iterating the files and moving them, if they end with `.log`. I think that would be more readable

Reply:
[Done.](https://github.com/MultiMC/MultiMC5/pull/3081/commits/8bb5ba76f114b69bfe349160929ef793f9151734) I also decided to iterate through all the data files to move, not just the logs.